File: programmers/124나라의 숫자.cpp
#include <string>
#include <vector>

using namespace std;

string solution(int n) {
    string answer = "";
    while(n)
    {
        if(n%3 ==0) {answer.insert(0,"4");n = n/3-1;}
        else{
            int temp = n%3;
            answer.insert(0,to_string(temp));
            n = n/3;
        }
    }
    return answer;
}

File: JAVA/Programmers/124나라의 숫자.java
class Solution {

    public String solution(int n) {
        String answer = "";
        StringBuilder result = new StringBuilder();

        while(true){
            if(n % 3 == 0){n/=3;n--; result.append("4");}
            else if(n % 3 == 1){n/=3; result.append("1");}
            else if(n % 3 == 2){n/=3; result.append("2");}
            if(n <= 0) break;
        }
        answer = result.reverse().toString();
        return answer;
    }
}
